guides/session.rst

```diff
@@ -1,76 +1,114 @@
 Controlling the Browser
 =======================
 
-Ok. Now we know how to create the browser driver to talk with a specific
-browser emulator. Although we can use drivers directly to call some actions
-on the emulator, Mink provides a better way - ``Session``:
+In Mink, the entry point to the browser is called the session. Think about
+it as being your browser window (some drivers even allow to deal with switching
+tabs).
+
+The first thing to do with your session is to start it. Nothing can be done
+with it before starting it.
 
 .. code-block:: php
 
-    // init session:
+    // choose a Mink driver. More about it in later chapters
+    $driver = new \Behat\Mink\Driver\GoutteDriver();
+
     $session = new \Behat\Mink\Session($driver);
 
-    // start session:
+    // start the session
     $session->start();
 
 .. note::
 
-  As you can see, the first argument to the session (``$driver``) is just
-  a simple driver instance, which we created in the previous chapter.
+    The first argument to the session constructor is a driver object, which
+    is how the Mink abstraction layer works. You will discover more about
+    the available drivers in a :doc:`later chapter </guides/drivers>`.
+
 
-``start()`` call is required in order to configure the browser emulator or
-controller to be fully functional.
+.. note::
+
+    Although Mink does its best on removing browser differences between different
+    browser emulators, it can't do much in some cases. See the :ref:`driver-feature-support`
+    to see which features are supported by each driver.
 
 Basic Browser Interaction
-~~~~~~~~~~~~~~~~~~~~~~~~~
+-------------------------
 
-After you've instantiated the ``$session`` object, you can control the actual
-browser emulator with it:
+The first thing to do to use your session is to open a page with it. Just
+after starting, the session is not on any page (in a real browser, you would
+on ``about:blank``), and calling any other action is likely to fail.
 
 .. code-block:: php
 
-    // open some page in browser:
     $session->visit('http://my_project.dev/some_page.php');
 
-    // get the current page URL:
-    echo $session->getCurrentUrl();
+.. note::
 
-    // get the response status code:
-    echo $session->getStatusCode();
+    Mink is primarily designed to be used for testing websites. To allow
+    covering error pages as well, ``Session::visit`` does not consider that
+    error status codes are invalid. It will not throw an exception in this
+    case. You will need to check whether the response was a success or an
+    error. It will only throw an exception when Mink cannot load the page
+    (network error, ...).
+
+Interacting with the Page
+-------------------------
+
+The session gives you access to the page through the ``Session::getPage``
+method. This allows you to :doc:`traverse it </guides/traversing-pages>` and
```

Comment:
Better use `it` one time at the end instead of using it in both links. Final text would look like `This allows you to [traverse] and [manipulate] it` (text in brackets is link).